lib/resty/etcd/v3.lua

```diff
@@ -144,7 +141,7 @@ local function _request_uri(self, method, uri, opts, timeout, ignore_auth)
             return nil, err
         end
     else
-        local max_retry = #self.endpoints * health_check.conf.max_fails - 1
+        local max_retry = #self.endpoints * health_check.conf.max_fails + 1
```

Comment:
"add another retry", should it be `#self.endpoints * health_check.conf.max_fails` instead?

Reply:
It's actually adding extra two retries 🤣

Before we take "-1" is due to
1. assumed there is one time of trial before getting into the retry loop, but there's not
2. forgot after all endpoints go unhealthy, we need an extra retry to get the error `has no healthy etcd endpoint available`